### PR TITLE
Activate TypeScript

### LINKED ISSUE #69: Introduce TypeScript
As part of the work on the Solid data browser (that we mentioned in https://github.com/solid/solid-panes/issues/64 earlier) we want to introduce TypeScript to improve the code quality and making future refactoring of the code base easier. We will work on this in a separate branch until we know that it is stable enough and have consensus from the community.

[TypeScript](https://www.typescriptlang.org/) offers a variety of features that makes working with JavaScript easier. We suggest migrating code file by file as we improve the codebase, and making sure to enhance the new files with types where needed (ie set `noImplicitAny` to `true`).

We need to make sure that the build setup is rigid but flexible enough so that developers who do not want to use TypeScript for their work can opt out.

### PR DESCRIPTION
**Pull request by Vinnl**
(Note: only relevant after #74 is merged.)

This PR enables type checking. The approach is as follows:

- There's a Babel plugin to remove type annotations from the code.
- In parallel to compiling the code, Fork-TS-Checker-Plugin (for Webpack) checks the type annotations.

I chose this approach because it appears we're already using Babel, and this makes it easy not to have type checking block compilation, which helps with gradual adoption.

TypeScript is currently set to strict mode, but is only active for `.ts` files. This means we can gradually migrate by renaming files, and simply disable type checking for specific values using `any` when it's too much work to introduce proper types.

Critique away! :)

Fixes #69.

### CODE REVIEW

**Review by megoth — APPROVED** [review]
In general I want to avoid type `unknown` as much as possible, but will allow it for this PR.

**Comment by Vinnl** [comment]
@megoth They're basically there to indicate "do not forget to properly type this later". Since they cannot be used until they get properly typed anyway, I figured this is a way to avoid typing values that are currently unused (hence speeding up the migration), especially considering that I do not know what the proper interface would be :) (See also the `PaneDefinition` interface, which I think is probably mostly correct, but is still the result of me guessing the interface based on other code and hence might not be entirely correct.)

In other words: I'd be happy for you to insert the correct types if you know them, but given that an incorrect type is probably worse than no type, I left it at this :)
